conn.go

```diff
@@ -0,0 +1,66 @@
+package tlsdialer
+
+import (
+	"errors"
+	"io"
+	"net"
+
+	"github.com/tarantool/go-tarantool/v2"
+)
+
+type tntConn struct {
+	net    net.Conn
+	reader io.Reader
+	writer writeFlusher
+}
+
+// writeFlusher is the interface that groups the basic Write and Flush methods.
+type writeFlusher interface {
+	io.Writer
+	Flush() error
+}
+
+// Addr makes tntConn satisfy the Conn interface.
+func (c *tntConn) Addr() net.Addr {
+	return c.net.RemoteAddr()
+}
+
+// Read makes tntConn satisfy the Conn interface.
+func (c *tntConn) Read(p []byte) (int, error) {
+	return c.reader.Read(p)
+}
+
+// Write makes tntConn satisfy the Conn interface.
+func (c *tntConn) Write(p []byte) (int, error) {
+	var (
+		l   int
+		err error
+	)
+
+	if l, err = c.writer.Write(p); err != nil {
+		return l, err
+	} else if l != len(p) {
+		return l, errors.New("wrong length written")
+	}
+	return l, nil
+}
+
+// Flush makes tntConn satisfy the Conn interface.
+func (c *tntConn) Flush() error {
+	return c.writer.Flush()
+}
+
+// Close makes tntConn satisfy the Conn interface.
+func (c *tntConn) Close() error {
+	return c.net.Close()
+}
+
+// Greeting makes tntConn satisfy the Conn interface.
+func (c *tntConn) Greeting() tarantool.Greeting {
+	return tarantool.Greeting{}
+}
+
+// ProtocolInfo makes tntConn satisfy the Conn interface.
+func (c *tntConn) ProtocolInfo() tarantool.ProtocolInfo {
+	return tarantool.ProtocolInfo{}
```

Comment:
What's about the actual ones?

Reply:
Actual `ProtocolInfo` is filled in the `ProtocolDialer`. Actual `Greeting` is filled in the `GreetingDialer`.

Reply:
No, we already done it:
https://github.com/tarantool/go-tlsdialer/blob/94b60638fb8b3cbe3c2514e5fd90d66780095119/dialer.go#L98-L126

Reply:
No, `OpenSslDialer` builds this Frankenstein itself, without user knowing:
https://github.com/tarantool/go-tlsdialer/blob/94b60638fb8b3cbe3c2514e5fd90d66780095119/dialer.go#L99-L123

But if you want to build your own dialer, then yes. You need to choose what dialer-helpers to use (as example, do you need authentication for your dialer or not), and build your custom dialer from blocks.